### DISCUSSION TITLE
How can I access current data of the updated collection in API hooks?

### DISCUSSION THREAD

**Discussion by G3root** [directus/directus#6382] [Archive]
In my collection I have a field which it's value is calculated during items before update lifecycle and depends on two other fields values , my question is how can I access the previous state of the current updated data if a depending field is updated.

**Comment by philleepflorence** [comment]
Are you are using `items.update.before` event? 
Have you tried the `payload` within the [Event Handler Function](https://docs.directus.io/guides/api-hooks/#event-handler-function)

**Reply by G3root** [reply]
That didn't help

**Comment by G3root** [comment]
I solved this problem by fetching the api

**Reply by philleepflorence** [reply]
Good to know you solved it.
